### PR TITLE
refactor: respect type for attribute builders

### LINKED ISSUE #1375: baseAttributeBuilder.type is not respected
`projectedAttributeBuilder` does not consider `baseAttribute.type()` when constructing the `projectedData`.

### PR DESCRIPTION
**Pull request by OliverForeman**
- Adds available types and associated lengths to `types`
- Adds `arrayViewFactory` to manage the typed array for each attribute
- Updates series to use appropriate types for each attribute

This resolves #1375

### CODE REVIEW

File: packages/d3fc-webgl/src/buffers/arrayViewFactory.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import typeMapper from './typeMapper';
+
+export default () => {
+    let ArrayType = Float32Array;
+    let cachedArray = new Float32Array(0);
```
Comment (chrisprice) [inline packages/d3fc-webgl/src/buffers/arrayViewFactory.js]: ’new ArrayType’

File: packages/d3fc-webgl/src/buffers/arrayViewFactory.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            cachedArray = new ArrayType(requiredLength);
+        }
+        return cachedArray;
+    };
+
+    factory.type = (...args) => {
+        if (!args.length) {
+            return ArrayType;
+        }
+        ArrayType = typeMapper(args[0]);
```
Comment (chrisprice) [inline packages/d3fc-webgl/src/buffers/arrayViewFactory.js:20]: What happens to the cached array?

**Review by chrisprice — CHANGES_REQUESTED** [review]
Sorry forgot to submit the review...

File: packages/d3fc-webgl/src/buffers/arrayViewFactory.js (hunk 2 above)
Comment (OliverForeman) [reply packages/d3fc-webgl/src/buffers/arrayViewFactory.js:20]: Added in a reset for the cached array

**Review by chrisprice — APPROVED** [review]

**Pull request merged**